src/utils.ts

```diff
@@ -78,17 +79,20 @@ export function findPagesDirectory({ nextRoot }: { nextRoot: string }) {
  * Default config:
  * https://github.com/vercel/next.js/blob/canary/packages/next/next-server/server/config.ts
  */
-function getNextConfig({ pathToConfig }: { pathToConfig: string }) {
-  return loadConfig(PHASE_DEVELOPMENT_SERVER, pathToConfig);
+export function getNextConfig({ pathToConfig }: { pathToConfig: string }) {
+  const config = loadConfig(PHASE_DEVELOPMENT_SERVER, pathToConfig);
+
+  setConfig(config);
```

Comment:
It's a bit confusing that `getter` mutates some "global" state and sets the config. Could we pull that out and let the callee execute it?

Reply:
Yeah good point, was avoiding moving too much around but agree it's confusing, updating!